src/main/java/com/aws/greengrass/componentmanager/KernelConfigResolver.java

```diff
@@ -416,6 +417,22 @@ private List<String> pointerToPath(JsonPointer pointer) {
         return new ArrayList<>();
     }
 
+    /**
+     * Interpolate the lifecycle commands or config with resolved system configuration values.
+     *
+     * @param configValue                 original value; could be Map or String
+     * @param componentIdentifier         target component id
+     * @param dependencies                name set of component's dependencies
+     * @param resolvedKernelServiceConfig resolved kernel configuration under "Services" key
+     * @return the interpolated lifecycle object
+     * @throws IOException for directory issues
+     */
+    public Object interpolateSystemParametersOnly(Object configValue, ComponentIdentifier componentIdentifier,
+                                                 Set<String> dependencies,
+                                                  Map<String, Object> resolvedKernelServiceConfig) throws IOException {
+        return interpolate(configValue, componentIdentifier, dependencies, resolvedKernelServiceConfig, false);
```

Comment:
I don't really like the idea of adding this boolean. Perhaps pass in the map of interpolatable things instead, and in this case copy the map and remove configuration.

Reply:
This boolean make it explicit that if it's interpolate all parameters or only the system parameters which is more deterministic and readable. If we copy the map and remove configuration, it's more dynamic and expose the interpolation behavior to the caller. What's the benefit of doing so?

Reply:
A free floating boolean is more difficult to understand than using an enum or even what I'm suggesting with the map. I expect that this change will be temporary, so I'm OK with this for now. If it stays longer, I'd like a more explicit option.